liquid-compiler/src/grammar.pest

```diff
@@ -29,9 +29,11 @@ Variable = ${ Identifier
               )*
             }
 Value = { Literal | Variable }
-Filter = { Identifier ~ (":" ~ Value ~ ("," ~ Value)*)? }
+Filter = { Identifier ~ (":" ~ FilterArgument ~ ("," ~ FilterArgument)*)? }
 FilterChain = { Value ~ ("|" ~ Filter)* }
-
+PositionalFilterArgument = {Value}
+KeywordFilterArgument = {Identifier ~ ":" ~ Value}
+FilterArgument = _{PositionalFilterArgument | KeywordFilterArgument}
```

Comment:
I leave it to you but I'd recommend holding off on actually implementing #92 to a separate PR to keep this one more focused.

Reply:
I prefer to include them already, because they will be important in determing the design of the code. For example, the `FilterArguments` function, or methods inside `FilterReflection` such as `positional_parameters` and `keyword_parameters`.

I fear that, if we hold off on them, there is a possibility that we overlook some important aspect that would then require rethinking and refactoring the API again (if we get this PR going just to change the API in the next PR, we will basically need to reimplement all filters twice and there are a lot of them).